### PR TITLE
feat(pre-commit): adds senstive data check like api keys and linting …

### LINKED ISSUE #4029: Add Sensitive Data Check And Linting Check
Adding Sensitive data check in pre-commit check will check whether the sensitive data have been accidently been included for example an api key.

While the linting check will check if the code follows correct linting format.

### PR DESCRIPTION
**Pull request by BLANKatGITHUB**
This will check for sensitive data and linting problems before commit .
fix #4029

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -30,3 +30,21 @@ repos:
     rev: 23.7.0
     hooks:
       - id: black
+
+  - repo: https://github.com/gitleaks/gitleaks
+      rev: v8.16.3
+    hooks:
+      - id: gitleaks
+
+  - repo: https://github.com/jumanjihouse/pre-commit-hooks
```
Comment (romange) [inline .pre-commit-config.yaml:39]: we usually do not have shell scripts in our PRs. Is it really relevant to us?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -30,3 +30,21 @@ repos:
...
+    rev: 3.0.0
+    hooks:
+      - id: shellcheck
+  - repo: https://github.com/pocc/pre-commit-hooks
+    rev: v1.3.5
+    hooks:
+      - id: cpplint
+  - repo: https://github.com/pylint-dev/pylint
```
Comment (romange) [inline .pre-commit-config.yaml:47]: we use `black` for py linting.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -30,3 +30,21 @@ repos:
...
+      - id: gitleaks
+
+  - repo: https://github.com/jumanjihouse/pre-commit-hooks
+    rev: 3.0.0
+    hooks:
+      - id: shellcheck
+  - repo: https://github.com/pocc/pre-commit-hooks
+    rev: v1.3.5
+    hooks:
+      - id: cpplint
```
Comment (romange) [inline .pre-commit-config.yaml:46]: these checks are running locally, to my knowledge, not on github. Have you tried running them on our codebase?

**Comment by BLANKatGITHUB** [comment]
fair enough I will remove them.
And just leave gitleaks for secret checks .

**Review by romange — APPROVED** [review]
Thanks!

**Pull request merged**